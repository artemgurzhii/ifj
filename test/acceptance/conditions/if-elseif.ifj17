Scope
dim a as integer
dim res as integer
a = 1
if a == 1 then
  res = 42
elseif a <> 0 then
  res = 24
elseif a <> 1 then
  res = 24
elseif a <> 4 then
  res = 24
end if
End Scope
