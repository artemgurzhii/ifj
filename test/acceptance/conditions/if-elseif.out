.IFJcode17
DEFVAR GF@a
DEFVAR GF@res
MOVE GF@a int@1
JUMPIFEQ RES_IF_1 GF@a int@1
JUMPIFNEQ RES_IF_2 GF@a int@0
JUMPIFNEQ RES_IF_3 GF@a int@1
JUMPIFNEQ RES_IF_4 GF@a int@4
LABEL RES_IF_1
MOVE GF@res int@42
JUMP END_IF
LABEL RES_IF_2
MOVE GF@res int@24
JUMP END_IF
LABEL RES_IF_3
MOVE GF@res int@24
JUMP END_IF
LABEL RES_IF_4
MOVE GF@res int@24
JUMP END_IF
LABEL END_IF
